invenio_rdm_records/cli.py

```diff
@@ -278,3 +280,176 @@ def rebuild_index():
     affs_service.rebuild_index(identity=system_identity)
 
     click.secho("Reindexed records and vocabularies!", fg="green")
+
+
+# CUSTOM FIELDS
+
+
+@click.group()
+def custom_fields():
+    """InvenioRDM custom fields commands."""
+
+
+@custom_fields.group()
+def records():
+    """InvenioRDM custom fields records commands."""
+
+
+# helper functions
+def _prepare_mapping(fields_names, available_fields):
+    """Prepare ES mapping properties for each field."""
+    available_fields = {field.name: field for field in available_fields}
+    fields = []
+    if fields_names:
+        for field_name in fields_names:
+            fields.append(available_fields.get(field_name))
+    else:
+        fields = available_fields.values()
+
+    properties = {}
+    for field in fields:
+        properties[f"custom_fields.{field.name}"] = field.mapping
```

Comment:
usual question about objects with inner fields...

Reply:
IIRC, the way Zach made it work was by making a custom `CF` class that would return the nested object in `field.mapping`. Then it will work. That means that for now we only support primitive types and lists of those. Custom objects can be added but require a bit of coding (a `CF` class is like 20-30 lines that can go to the cfg file or to a custom module)